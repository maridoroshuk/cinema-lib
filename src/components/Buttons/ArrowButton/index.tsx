import React from 'react';

import ArrowImage from '@/assets/icons/arrow.png';

import { Arrow, Wrapper } from './styles';
import { IArrowButton } from './interface';

export const ArrowButton = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Arrow src={ArrowImage} alt="arrow (hide)" />
    </Wrapper>
  );
};
