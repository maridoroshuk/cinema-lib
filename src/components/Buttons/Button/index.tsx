import React from 'react';

import { ButtonStyled } from '@/components/Buttons/Button/styles';

import { IButtonProps } from './interface';

export const Button = ({
  children,
  size,
  backgroundColor,
  type,
  value,
  onClick,
  dataCy,
}: IButtonProps) => {
  return (
    <ButtonStyled
      backgroundColor={backgroundColor}
      size={size}
      type={type || 'button'}
      value={value}
      onClick={onClick}
      data-cy={dataCy}
    >
      {children}
    </ButtonStyled>
  );
};
