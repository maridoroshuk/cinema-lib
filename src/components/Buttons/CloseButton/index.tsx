import React from 'react';
import { ICloseButtonProps } from './interface';
import { ButtonWrapper, StyledButton } from './styles';

export const CloseButton = ({ onClose }: ICloseButtonProps) => {
  return (
    <ButtonWrapper>
      <StyledButton onClick={onClose} />
    </ButtonWrapper>
  );
};
