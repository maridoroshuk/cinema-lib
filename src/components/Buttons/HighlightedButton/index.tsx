import React from 'react';
import { IUnderlineButton } from './interface';
import { ButtonStyled } from './styles';

export const HighlightedButton = ({
  children,
  handleButtonClick,
  value,
}: IUnderlineButton) => {
  return (
    <ButtonStyled onClick={handleButtonClick} value={value || ''}>
      {children}
    </ButtonStyled>
  );
};
