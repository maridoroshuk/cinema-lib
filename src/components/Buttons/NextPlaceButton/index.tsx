import React from 'react';

import NextPlaceButton from '@/assets/icons/next_place_button.png';

import { Next, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const NextPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Next src={NextPlaceButton} />
    </Wrapper>
  );
};
