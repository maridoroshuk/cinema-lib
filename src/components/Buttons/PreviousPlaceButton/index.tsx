import React from 'react';

import PreviousPlaceButton from '@/assets/icons/previous_place_button.png';

import { Previous, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const PreviousPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Previous src={PreviousPlaceButton} />
    </Wrapper>
  );
};
