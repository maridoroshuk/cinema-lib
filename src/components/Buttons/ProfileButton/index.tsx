import React from 'react';
import { IProfileButton } from './interface';
import { ButtonStyled } from './styles';

export const ProfileButton = ({ onClick }: IProfileButton) => {
  return <ButtonStyled onClick={onClick}>Profile</ButtonStyled>;
};
