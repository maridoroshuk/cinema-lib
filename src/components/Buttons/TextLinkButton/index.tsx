import React from 'react';
import { ITextLinkButtonProps } from './interface';
import { StyledLink } from './styles';

export const TextLinkButton = ({
  children,
  value,
  onClick,
}: ITextLinkButtonProps) => {
  return (
    <StyledLink value={value} onClick={onClick}>
      {children}
    </StyledLink>
  );
};
