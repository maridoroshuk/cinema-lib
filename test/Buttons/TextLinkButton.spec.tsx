import React from 'react';
import { render } from '@testing-library/react';

import { TextLinkButton } from '@/components/Buttons/TextLinkButton';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('TextLinkButton', () => {
  it('renders with signUp value', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <TextLinkButton onClick={handleClick} value="signUp">
          Sign up please.
        </TextLinkButton>,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with logIn value', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <TextLinkButton onClick={handleClick} value="logIn">
          Login please.
        </TextLinkButton>,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
