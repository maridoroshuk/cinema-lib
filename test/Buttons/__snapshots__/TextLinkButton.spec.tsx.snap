// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLinkButton renders with logIn value 1`] = `
.c0 {
  font-style: italic;
  font-size: 20px;
  color: #FFFFFF;
  margin-left: 10px;
  border: none;
  outline: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  background-color: transparent;
}

<button
  class="c0"
  value="logIn"
>
  Login please.
</button>
`;

exports[`TextLinkButton renders with signUp value 1`] = `
.c0 {
  font-style: italic;
  font-size: 20px;
  color: #FFFFFF;
  margin-left: 10px;
  border: none;
  outline: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  background-color: transparent;
}

<button
  class="c0"
  value="signUp"
>
  Sign up please.
</button>
`;
